jobs/business-sync/Dockerfile

```diff
@@ -62,9 +65,11 @@ RUN groupadd -g "${GID}" -r web \
   && useradd -d '/code' -g web -l -r -u "${UID}" web \
   && chown web:web -R '/code'
 
+RUN mkdir /cloudsql && chown web:web /cloudsql
```

Comment:
is anything copied to this directory? have you tried running Dockerfile locally, to make sure cloudsql proxy binary is there and is executable?

Reply:
I build an image from your Dockerfile and inspected the files - there is no cloud_sql_proxy installed in the image.

In the Dockerfile you've provided, the cloud_sql_proxy binary is being downloaded and placed in /usr/local/bin/ in the development_build stage:
curl -o /usr/local/bin/cloud-sql-proxy https://storage.googleapis.com/cloud-sql-connectors/cloud-sql-proxy/v2.1.0/cloud-sql-proxy.linux.amd64 \
&& chmod +x /usr/local/bin/cloud-sql-proxy

However, the issue lies in how multi-stage builds work. When you use:

FROM development_build AS production_build
This starts a new production_build stage, which only copies the /code directory from the previous stage (development_build), but does not carry over the other files, like the cloud_sql_proxy binary that was installed in /usr/local/bin/.

So /cloudsql directory is empty. 

You either need to copy cloud_sql_proxy binary between stages:

FROM development_build AS production_build

# Copy necessary files for production
COPY --chown=web:web . /code

# Ensure cloud_sql_proxy is included in the final image
COPY --from=development_build /usr/local/bin/cloud-sql-proxy /usr/local/bin/cloud-sql-proxy

USER web

# Start the application
CMD [ "/bin/sh", "/code/run.sh" ]

or move installation to the second stage:

FROM development_build AS production_build

# Copy necessary files for production
COPY --chown=web:web . /code

# Install cloud_sql_proxy again in the production build
USER root
RUN curl -o /usr/local/bin/cloud-sql-proxy https://storage.googleapis.com/cloud-sql-connectors/cloud-sql-proxy/v2.1.0/cloud-sql-proxy.linux.amd64 \
  && chmod +x /usr/local/bin/cloud-sql-proxy

USER web

# Start the application
CMD [ "/bin/sh", "/code/run.sh" ]

Reply:
Thanks for the advise, just updated